src/main/java/org/opentripplanner/model/plan/Leg.java

```diff
@@ -149,9 +150,7 @@
     * For transit legs, the service date of the trip.
     * For non-transit legs, null.
     */
-   // TODO OTP2 - This should not be a String? What is this used for? Is it the actual date or the
-   //           - Service date?
-   public String serviceDate = null;
+   public ServiceDate serviceDate = null;
```

Comment:
I think we should add more JavaDoc here and maybe also rename the field to `tripServiceDate`. Add doc (suggestion):
```
The trip service date should be used to identify the correct trip schedule and can not be trusted to display the date for any departures or arrivals. For example, the first departure for a given trip may happen at service date March 25th and service time 25:00, which in local time would be Mach 26th 01:00.  
```